language_detector/main.py

```diff
@@ -2,8 +2,52 @@
 
 from .languages import LANGUAGES
 
+def load_languages(language_list=LANGUAGES):
+    # Input: A list of dicts where within each dict in the list, the value for the key 
+    #   'name' is the language and the value key 'common words' is a list of common words. 
+    # Returns:  A dict with the key being the names of the lanauges and the values 
+    #   a int initialized to zero.
+    
+    result = {}
+    for lang_dict in language_list:
+        result[lang_dict.get('name')] = 0
+    return result
+
+#assert load_languages() == {'German': 0, 'Spanish': 0}
+
 
 def detect_language(text, languages=LANGUAGES):
-    """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+    # Input: A string of text with unknown language, 'text', and a list of dicts, 'langauges' 
+    #    where within each dict in the list, the value for the key 'name' is the language and 
+    #    the value key 'common words' is a list of common words.
+    # Returns: The detected language of given text as a string.
+    
+    # Pre-load our results dictionary:
+    results_dict = load_languages(languages)
+    
+    for lang_dict in languages:
+        langname = lang_dict['name']
+        wordlist = lang_dict['common_words']
+        results_dict[langname] = len([word for word in text.lower().split() if (word in wordlist)])
```

Comment:
We used a counter instead of length, but this is a really smart way of doing it. Only problem I see is how you handle special characters. For example in English, hyphenated words like "runner-up" would most likely never be in a common words list, but "up" might be. This could throw you off in a mixed language paragraph.